api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -0,0 +1,200 @@
+  implicit val materializer: ActorMaterializer
+  val log: LoggingAdapter
+
+  implicit val timeout: Timeout
+
+  val splitLines = Framing.delimiter(ByteString("\n"), 2048, allowTruncation = true)
+
+  val institutionsPath =
+    path("institutions") {
+      val institutionsActor = system.actorSelection("/user/institutions")
+      get {
+        val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
+        onComplete(fInstitutions) {
+          case Success(institutions) =>
+            complete(ToResponseMarshallable(Institutions(institutions)))
+          case Failure(error) =>
+            log.error(error.getLocalizedMessage)
+            complete(HttpResponse(StatusCodes.InternalServerError))
+        }
+      }
+    }
+
+  val institutionByIdPath =
+    path("institutions" / Segment) { fid =>
+      extractExecutionContext { executor =>
+        val institutionsActor = system.actorSelection("/user/institutions")
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fInstitutionDetails = institutionDetails(fid, institutionsActor, filingsActor)
+          onComplete(fInstitutionDetails) {
+            case Success(institution) =>
+              filingsActor ! Shutdown
+              complete(ToResponseMarshallable(institution))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              log.error(error.getLocalizedMessage)
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val filingByPeriodPath =
+    path("institutions" / Segment / "filings" / Segment) { (fid, period) =>
+      extractExecutionContext { executor =>
+        val filingsActor = system.actorOf(FilingPersistence.props(fid))
+        val submissionActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        get {
+          implicit val ec: ExecutionContext = executor
+          val fDetails: Future[FilingDetail] = filingDetailsByPeriod(period, filingsActor, submissionActor)
+          onComplete(fDetails) {
+            case Success(filingDetails) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(ToResponseMarshallable(filingDetails))
+            case Failure(error) =>
+              filingsActor ! Shutdown
+              submissionActor ! Shutdown
+              complete(HttpResponse(StatusCodes.InternalServerError))
+          }
+        }
+      }
+    }
+
+  val submissionPath =
+    path("institutions" / Segment / "filings" / Segment / "submissions") { (fid, period) =>
+      val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+      post {
+        val submissionsActor = system.actorOf(SubmissionPersistence.props(fid, period))
+        submissionsActor ! CreateSubmission
+        val fLatest = (submissionsActor ? GetLatestSubmission).mapTo[Submission]
```

Comment:
If there are multiple users trying to create submissions are we guaranteed that the submission that was just created will be the latest? Say if there are two users trying to create a submission for the institution, is it possible that 1 will send `CreateSubmission` then 2 will send `CreateSubmission` then 1 will send `GetLatestSubmission`. If this happened then 1's `GetLatestSubmission` would return the submission created by 2 not by 1. I will admit that this is a small corner case but it might still be good to address.